### PR TITLE
Fixes: Publicize

### PR DESCRIPTION
**Pull request by enejb**
- Make sure we sync post meta that is needed by publicize and other
Jetpack Modules.
- Also 2 fixes publicize by making sure we sync post meta related to them. 
Skip publicize for some accounts `_wpas_skip_*`
Publicize as well as custom message. `_wpas_mess`

cc @gravityrail, @lezama

### CODE REVIEW

**Comment by gravityrail** [comment]
lgtm! :shipit:

File: sync/class.jetpack-sync-defaults.php [hunk 1]
```diff
@@ -202,7 +215,7 @@ static function is_whitelisted_option( $option ) {
 				return true;
 			}
 		}
-
```
Comment (ebinnion) [inline sync/class.jetpack-sync-defaults.php]: I'm mobile, but it looks like there is extra whitespace at the beginning of this line.

File: sync/class.jetpack-sync-defaults.php [hunk 2]
```diff
@@ -202,7 +215,7 @@ static function is_whitelisted_option( $option ) {
...
+
```
Comment (ebinnion) [inline sync/class.jetpack-sync-defaults.php]: I'm mobile, but it looks like there's extra whitespace at the beginning of this line.

**Comment by enejb** [comment]
@samhotchkiss 
Cherry picked to branch-4.2 https://github.com/Automattic/jetpack/commit/43add04a66ffa25ea4ba6c1bc594a6254d1968bc

**Pull request merged**